File: java-packages/src/main/java/com/rmaciak/order/domain/OrderEntity.java
package com.rmaciak.order.domain;

import java.math.BigDecimal;
import java.util.Set;
import java.util.UUID;

import static java.util.stream.Collectors.toSet;

public record OrderEntity(UUID id, BigDecimal quota, Set<OrderItemEntity> items) {

    static OrderEntity from(Order order) {
        return new OrderEntity(
                order.orderId(),
                order.orderQuota(),
                order.orderItems().stream().map(OrderItemEntity::from).collect(toSet())
        );
    }

    Order toDomain() {
        return new Order(
                id,
                quota,
                items.stream().map(OrderItemEntity::toDomain).collect(toSet())
        );
    }
}

File: kotlin-with-submodules/order-domain/src/main/kotlin/com/rmaciak/order/OrderEntity.kt
package com.rmaciak.order

import java.math.BigDecimal
import java.util.UUID

data class OrderEntity(
        val id: UUID,
        val quota: BigDecimal,
        val items: Set<OrderItemEntity>
)

data class OrderItemEntity(
        val id: UUID,
        val quota: BigDecimal
)
